src/test/osd/TestMClockScheduler.cc

```diff
@@ -169,3 +181,31 @@ TEST_F(mClockSchedulerTest, TestMultiClientOrderedEnqueueDequeue) {
   }
   ASSERT_TRUE(q.empty());
 }
+
+TEST_F(mClockSchedulerTest, TestHighPriorityQueueEnqueueDequeue) {
```

Comment:
Test looks good. It would be good to have additional test(s) that creates a mix of immediate, high_priority and mclock scheduler items and test if they get dequeued in the expected way. The idea is ensure maximum coverage possible of the code that is introduced.